MergerLogic/DataTypes/Data.cs

```diff
@@ -173,7 +235,7 @@ public static Data CreateDatasource(string type, string path, int batchSize, boo
             return data;
         }
 
-        public static Data CreateDatasource(string type, string path, int batchSize, bool isBase, Extent extent, int maxZoom, int minZoom = 0, bool isOneXone = false)
+        public static Data CreateDatasource(string type, string path, int batchSize, bool isBase, Extent extent, int maxZoom, int minZoom = 0, bool isOneXone = false, TileGridOrigin? origin = null)
```

Comment:
why it's null? why you don't set the origin to a default on startup and prevent all the "if null" checks?